go/database/mpt/nodes.go

```diff
@@ -301,6 +301,7 @@ func visitPathTo(source NodeSource, root *NodeReference, path []Nibble, address
 			if n.path.IsPrefixOf(path) {
 				nodeId = &n.next
 				path = path[n.path.Length():]
+				done = len(path) == 0
```

Comment:
This should actually not be possible in a consistent tree. Also, for BranchNodes, the path should never be fully consumed. However, I guess it does not harm to have this code here.